compiler/ballerina-lang/src/main/java/io/ballerina/projects/PackageResolution.java

```diff
@@ -485,14 +486,41 @@ private BlendedManifest createBlendedManifest(PackageContext rootPackageContext,
 
     private ResolutionOptions getResolutionOptions(PackageContext rootPackageContext,
                                                    CompilationOptions compilationOptions) {
+        boolean sticky = getSticky(rootPackageContext);
+        PackageLockingMode packageLockingMode;
+        SemanticVersion prevDistributionVersion = rootPackageContext.dependencyManifest().distributionVersion();
+        SemanticVersion currentDistributionVersion = SemanticVersion.from(RepoUtils.getBallerinaShortVersion());
+        if (!sticky && (null == prevDistributionVersion
+                || isMinorOrMajorVersionIncrement(prevDistributionVersion, currentDistributionVersion))) {
+            packageLockingMode = PackageLockingMode.SOFT;
+            DiagnosticInfo diagnosticInfo = new DiagnosticInfo(
+                    DiagnosticWarningCode.BUILT_WITH_OLDER_SL_UPDATE_DISTRIBUTION.diagnosticId(),
+                    "WARNING: Detected packages built with an older SL Update distribution. " +
+                            "Dependencies will be updated to the latest versions compatible " +
+                            "with the current distribution.",
```

Comment:
@sameerajayasoma can this warning be improved?

Reply:
Detected an attempt to build this package using Swn Lake Update 5. However, this package was built using Swan Lake Update 4.1. To ensure compatibility, the `Dependencies.toml` file will be updated with the latest versions that are compatible with Update 5.

Reply:
So the warning message would be in the format,
> Detected an attempt to build this package using Swn Lake Update `<minor>.<patch>`. However, this package was built using Swan Lake Update `<minor>.<patch>`. To ensure compatibility, the Dependencies.toml file will be updated with the latest versions that are compatible with Update `<minor>.<patch>`.

Also in cases where Dependencies.toml doesn't contain `distribution-version` (update 4 or below), @azinneera suggested to use `...was built using Swan Lake Update 4 or below`. 

Hope it is fine.

Reply:
Let reword `below` to `an older Update`